src/page-modules/assistant/trip/trip-pattern/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { useClientWidth } from '@atb/utils/use-client-width';
+import { motion } from 'framer-motion';
+import { Fragment, useEffect, useState } from 'react';
+import { getFilteredLegsByWalkOrWaitTime, tripSummary } from './utils';
+import { PageText, useTranslation } from '@atb/translations';
+import { type TripPattern as TripPatternType } from '../../server/journey-planner/validators';
+import style from './trip-pattern.module.css';
+import { formatLocaleTime, isInPast } from '@atb/utils/date';
+import { TripPatternHeader } from './trip-pattern-header';
+import { TransportIconWithLabel } from '@atb/components/transport-mode/transport-icon';
+import { MonoIcon } from '@atb/components/icon';
+import { Typo } from '@atb/components/typography';
+
+const LAST_LEG_PADDING = 20;
+
+type TripPatternProps = {
+  tripPattern: TripPatternType;
+  delay: number;
+  index: number;
+};
+
+export default function TripPattern({
+  tripPattern,
+  delay,
+  index,
+}: TripPatternProps) {
+  const { t, language } = useTranslation();
+
+  const filteredLegs = getFilteredLegsByWalkOrWaitTime(tripPattern);
+
+  const [numberOfExpandedLegs, setNumberOfExpandedLegs] = useState(
+    filteredLegs.length,
+  );
+
+  const expandedLegs = filteredLegs.slice(0, numberOfExpandedLegs);
+  const collapsedLegs = filteredLegs.slice(
+    numberOfExpandedLegs,
+    filteredLegs.length,
+  );
+
+  const [legsParentWidth, legsParentRef] = useClientWidth<HTMLDivElement>();
+  const [legsContentWidth, legsContentRef] = useClientWidth<HTMLDivElement>();
+
+  // Dynamically collapse legs to fit horizontally
+  useEffect(() => {
+    if (legsParentWidth && legsContentWidth) {
+      if (legsContentWidth >= legsParentWidth - LAST_LEG_PADDING) {
```

Comment:
Added a `LAST_LEG_PADDING` here to make some space between the destination leg and second to last leg. Couldn't do it in a nice way with CSS without sacrificing the line in some way.

Reply:
I'm not seeing a better approach right now, so I'm fine with this 👍